astropy/units/quantity.py

```diff
@@ -1303,15 +1303,13 @@ def __setitem__(self, i, value):
     # __contains__ is OK
 
     def __bool__(self):
-        """Quantities should always be treated as non-False; there is too much
-        potential for ambiguity otherwise.
+        """This method raises ValueError, since truthiness of quantities is ambiguous,
+        especially for logarithmic units and temperatures. Use explicit comparisons.
         """
-        warnings.warn(
-            "The truth value of a Quantity is ambiguous. "
-            "In the future this will raise a ValueError.",
-            AstropyDeprecationWarning,
+        raise ValueError(
+            "Truthiness of quantities is ambiguous, especially for logarithmic units"
```

Comment:
Misspelled "ambiguous" here, hence the failures.

Also, super minor but while you are at it: exception messages start with lower case in CPython (as they follow a colon, I guess) and we've been trying (haphazardly) to follow that.

(Looks good otherwise)